pkg/deploy/lattice/target_group_manager.go

```diff
@@ -392,6 +316,55 @@ func (s *defaultTargetGroupManager) findTargetGroup(
 	return nil, nil
 }
 
+// latticeTags will be fetched if nil
+func (s *defaultTargetGroupManager) IsTargetGroupMatch(ctx context.Context,
+	modelTg *model.TargetGroup, latticeTg *vpclattice.TargetGroupSummary,
+	latticeTagsAsModelTags *model.TargetGroupTagFields) (bool, error) {
+
+	// first check fields we have before we try tags
+	if aws.Int64Value(latticeTg.Port) != int64(modelTg.Spec.Port) ||
+		aws.StringValue(latticeTg.Protocol) != modelTg.Spec.Protocol ||
+		aws.StringValue(latticeTg.IpAddressType) != modelTg.Spec.IpAddressType ||
+		aws.StringValue(latticeTg.Type) != string(modelTg.Spec.Type) ||
+		aws.StringValue(latticeTg.VpcIdentifier) != modelTg.Spec.VpcId {
+
+		return false, nil
+	}
+
+	// so far so good, now we check tags
+	if latticeTagsAsModelTags == nil {
+		// fetch the tags if we don't have them already
+		req := vpclattice.ListTagsForResourceInput{ResourceArn: latticeTg.Arn}
+		res, err := s.cloud.Lattice().ListTagsForResourceWithContext(ctx, &req)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				// may have been deleted in the meantime, that's OK
+				return false, nil
+			}
+
+			return false, err
+		}
+
+		tags := model.TGTagFieldsFromTags(res.Tags)
+		latticeTagsAsModelTags = &tags
+	}
+
+	tagsMatch := model.TagFieldsMatch(modelTg.Spec, *latticeTagsAsModelTags)
+	if !tagsMatch {
+		return false, nil
+	}
+
+	// one last check - ProtocolVersion is not present on TargetGroupSummary, so we have to do a Get
```

Comment:
I'm curious if we could save a call by putting protocolversion to tags.

Reply:
I actually think we should just return it in the API as part of List